src/cogent3/app/sqlite_data_store.py

```diff
@@ -295,6 +277,64 @@ def drop_not_completed(self) -> None:
         self.db.execute(f"DELETE FROM {_RESULT_TABLE} WHERE is_completed=0")
         self._not_completed = []
 
+    def lock(self):
+        """if writable, and not locked, locks the database to this pid"""
+        # if mode=w and the data store exists AND has a lock_pid
+        # value already, then we should fail. The user might
+        # inadvertently overwrite something otherwise.
+        # BUT if mode=a, as the user expects to be modifying the data
+        # store then we have to update the value
+        if self.mode is not READONLY:
+            result = (
+                None
+                if self.mode is APPEND
+                else self._db.execute("SELECT state_id,lock_pid FROM state")
+            ).fetchall()
+            unlocked = False
```

Comment:
`unlocked` seems pointless

Reply:
The problem is to distinguish between 1- We don't have that record in `state` table, or 2- We have it and `lock_pid` field of that record is NULL.

Reply:
@GavinHuttley 
I cannot make `not_locked` tinydb on demand, because we will remove tinydb support and I cannot change a tinydb file in program. I am using this file to test converting tinydb to sqlitedb. Now I added a new test for converting not_locked tinydb.

Reply:
we can change the locked tinydb:
- int a new fixture, do the same imports as done in the `convert_tiny...` function
- copy locked file into a temp directory
- load tinydb
- remove the "LOC" record
- flush to disk
- etc...

Reply:
OK

Reply:
@GavinHuttley 
I know this method, but I didn't want to use anything from tinydb imports.
OK, I am going to change it.